tox.ini

```diff
@@ -319,68 +323,143 @@ setenv =
   CORE_REPO=git+https://github.com/open-telemetry/opentelemetry-python.git@{env:CORE_REPO_SHA}
 
 commands_pre =
-; Install without -e to test the actual installation
-  py3{8,9,10,11}: python -m pip install -U pip setuptools wheel
-; Install common packages for all the tests. These are not needed in all the
-; cases but it saves a lot of boilerplate in this file.
-  test: pip install opentelemetry-api@{env:CORE_REPO}\#egg=opentelemetry-api&subdirectory=opentelemetry-api
```

Comment:
Didn't we have this here to remove the exact boiler plate code below? I know it is more explicit for what dependencies are installed per environment but is that better than repeated the same install lines? Open to discussing it during this weeks sig.

Reply:
Yes, my plan is to fragment the `tox.ini` file in one per package so that we don't have this huge file, this PR is a step in that direction. That hopefully also would make `lint` faster by taking advantage of paralellization in CI and would allow the user to only have to run `lint` for the packages that the user changes.